src/compile/common.ts

```diff
@@ -114,7 +117,10 @@ export function formatMixins(model: Model, channel: Channel, format: string) {
         def.format = model.config().numberFormat;
         break;
       case TEMPORAL:
-        def.format = timeFormat(model, channel) || model.config().timeFormat;
+        if ((!fieldDef.timeUnit) || (fieldDef.timeUnit &&
+              fieldDef.timeUnit.toString().indexOf('quarter') === -1)) {
```

Comment:
the second argument should be a timeUnit so you will need to convert it to a string. Let me know if you can't find out how to convert from a ts enum value to a string.